transports/websocket/src/lib.rs

```diff
@@ -18,66 +18,182 @@
-//!
-//! See the documentation of `swarm` and of libp2p in general to learn how to use the `Transport`
-//! trait.
-//!
-//! This library is used in a different way depending on whether you are compiling for emscripten
-//! or for a different operating system.
-//!
-//! # Emscripten
-//!
-//! On emscripten, you can create a `BrowserWsConfig` object with `BrowserWsConfig::new()`. It can
-//! then be used as a transport.
-//!
-//! Listening on a websockets multiaddress isn't supported on emscripten. Dialing a multiaddress
-//! which uses `ws` on top of TCP/IP will automatically use the `XMLHttpRequest` Javascript object.
-//!
-//! ```ignore
-//! use libp2p_websocket::BrowserWsConfig;
-//!
-//! let ws_config = BrowserWsConfig::new();
-//! // let _ = ws_config.dial("/ip4/40.41.42.43/tcp/12345/ws".parse().unwrap());
-//! ```
-//!
-//! # Other operating systems
-//!
-//! On other operating systems, this library doesn't open any socket by itself. Instead it must be
-//! plugged on top of another implementation of `Transport` such as TCP/IP.
-//!
-//! This underlying transport must be put inside a `WsConfig` object through the
-//! `WsConfig::new()` function.
-//!
-//! ```
-//! extern crate libp2p_core;
-//! extern crate libp2p_tcp;
-//! extern crate libp2p_websocket;
-//!
-//! use libp2p_core::{Multiaddr, Transport};
-//! use libp2p_tcp::TcpConfig;
-//! use libp2p_websocket::WsConfig;
-//!
-//! # fn main() {
-//! let ws_config = WsConfig::new(TcpConfig::new());
-//! # return;
-//! let _ = ws_config.dial("/ip4/40.41.42.43/tcp/12345/ws".parse().unwrap());
-//! # }
-//! ```
-//!
-
-#[cfg(any(target_os = "emscripten", target_os = "unknown"))]
-#[macro_use]
-extern crate stdweb;
-
-#[cfg(any(target_os = "emscripten", target_os = "unknown"))]
-mod browser;
-#[cfg(not(any(target_os = "emscripten", target_os = "unknown")))]
-mod desktop;
-
-#[cfg(any(target_os = "emscripten", target_os = "unknown"))]
-pub use self::browser::{BrowserWsConfig, BrowserWsConn};
-#[cfg(not(any(target_os = "emscripten", target_os = "unknown")))]
-pub use self::desktop::WsConfig;
+
+pub mod error;
+pub mod framed;
+pub mod tls;
+
+use error::Error;
+use framed::BytesConnection;
+use futures::prelude::*;
+use libp2p_core::{
+    ConnectedPoint,
+    Transport,
+    multiaddr::Multiaddr,
+    transport::{map::{MapFuture, MapStream}, ListenerEvent, TransportError}
+};
+use rw_stream_sink::RwStreamSink;
+use tokio_io::{AsyncRead, AsyncWrite};
+
+/// A Websocket transport.
+#[derive(Debug, Clone)]
+pub struct WsConfig<T> {
+    transport: framed::WsConfig<T>
+}
+
+impl<T> WsConfig<T> {
+    /// Create a new websocket transport based on the given transport.
+    pub fn new(transport: T) -> Self {
+        framed::WsConfig::new(transport).into()
+    }
+}
+
+impl<T> From<framed::WsConfig<T>> for WsConfig<T> {
+    fn from(framed: framed::WsConfig<T>) -> Self {
+        WsConfig {
+            transport: framed
+        }
+    }
+}
+
+impl<T> AsRef<framed::WsConfig<T>> for WsConfig<T> {
```

Comment:
Considering that the `WsConfig` here is supposed to be the mostly-used one, I think that all of its methods should simply be copy-pasted here as well. We want to make the user believe that they are two different types.